packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -520,7 +527,10 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
   /// The default implementation defers to [ScrollPhysics.adjustPositionForNewDimensions].
   @protected
   bool correctForNewDimensions(ScrollMetrics oldPosition, ScrollMetrics newPosition) {
-    final double newPixels = physics.adjustPositionForNewDimensions(
+    final double newPixels = activity.adjustPositionForNewDimensions(
+      oldPosition: oldPosition,
+      newPosition: newPosition,
+    ) ?? physics.adjustPositionForNewDimensions(
```

Comment:
Isn't this a little surprising that in certain situations the opinion of the physics gets ignored?

Reply:
Maybe we should at least document this on the physics's adjustPositionForNewDimensions that it may be overridden by the activity?

Reply:
Yes, I definitely want to improve the docs here if we go this route.

I tried to figure out a way to have both the activity and the physics contribute but I couldn't come up with a way. Fundamentally there's one number. I mean, I could take the mean or some such but...